layers/osi.go

```diff
@@ -0,0 +1,74 @@
+package layers
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/gopacket"
+)
+
+// For reference use Wireshark OSI dissector
+// https://github.com/wireshark/wireshark/blob/master/epan/dissectors/packet-osi.c
+// https://github.com/wireshark/wireshark/blob/master/epan/dissectors/packet-osi.h
+
+type OSI struct {
+	BaseLayer
+
+	// This type is defined in layers/enums.go
+	// All the methods for this type are auto-generated
+	// (see layers/enums_generated.go and layers/gen2.go for more details)
+	OSIType OSIType
+
+	Protocol byte
+	// TODO Describe other header fields
+}
+
+func (osi *OSI) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+
+	if len(data) < 1 {
+		return errors.New("OSI header too small")
+	}
+
+	osi.Protocol = data[0]
+
+	osi.Contents = data[:1]
+	osi.Payload = data[1:]
+
+	return nil
+}
+
+// LayerType returns LayerTypeISIS
+func (osi *OSI) LayerType() gopacket.LayerType {
+	return LayerTypeOSI
+}
+
+// NextLayerType returns the layer type contained by this DecodingLayer.
+func (osi *OSI) NextLayerType() gopacket.LayerType {
+
+	switch osi.Protocol {
```

Comment:
Instead of code on 48-55 lines you can use simple 
`return osi.OSIType.LayerType()`